file.go

```diff
@@ -118,7 +119,9 @@ func (f *win32File) closeHandle() {
 	if !f.closing.swap(true) {
 		// cancel all IO and wait for it to complete
 		cancelIoEx(f.handle, nil)
+		f.wgLock.Lock()
```

Comment:
>f.wgLock.Lock() [](start = 2, length = 15)

I expected this lock to be around the modification of f.closing, not the wait on the wait group... With this, a new IO on a closing handle will stall until all existing IOs complete.